### PR TITLE
Node update for compatibility with lts versions 18 and 20

### PR DESCRIPTION
**Pull request by Ricokola**
Update node for compatibility with lts versions 18 and 20.

### CODE REVIEW

**Review by Garneauma — APPROVED** [review]
Tested locally using Node v18.19.1 and it works correctly.

To do:
- [ ] Rebase and fix conflict
- [ ] Update the reference to WET-BOEW in package.json to the latest version when it is released.

**Comment by duboisp** [comment]
Equivalent change was merged in wet-boew, please rebase and update the build configuration as intended on the wet-boew tag is created.

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
   "license": "MIT",
   "dependencies": {
     "bootstrap-sass": "3.4.1",
-    "wet-boew": "github:wet-boew/wet-boew#v4.0.81.1"
+    "wet-boew": "github:Ricokola/wet-boew#node-sass4"
```
Comment (duboisp) [inline package.json:16]:
```suggestion
    "wet-boew": "github:wet-boew/wet-boew#v4.0.81.3"
```

**Review by duboisp — CHANGES_REQUESTED** [review]
The version of wet-boew that are running with Node 18-20 is now released. You can update your PR to align all the dependencies as expected.

**Comment by duboisp** [comment]
Well, after this change we will need to look for alternative for node-sass at our next upgrade of NodeJS to build GCWeb

image

**Comment by duboisp** [comment]
@Ricokola FYI, please review the step bellow and let's include those in our release note under "Building GCWeb developer note" sub-title.

Here my step that I took to upgrade my dev build environment on Window 10:

1. Uninstall existing NodeJS version if installed OS wide
2. Get NodeJS LTS recent version
   a. Preferably - Install NVM - Node Version Manager and use Node 20 LTS
   b. Or, install Node 20 LTS OS wide
4. Delete the folder "node-sass" under the "node_modules"
5. Run "script/setup" which would reinstall bower and install all the NodeJS libraries

Then, it will be as usual step to build
1. If wet-boew dependency is updated `> npm install`
2. Build GCWeb `> grunt`
3. Run Jekyll local website `> docker-compose up`
4. Test the webiste + code by navigating locally: http://localhost:4000/ 

---

There is an error in wet-boew, we need to change how we enforce the npm dependencies
```
npm warn EBADENGINE Unsupported engine {
npm warn EBADENGINE   package: 'wet-boew@4.0.81.3',
npm warn EBADENGINE   required: { npm: '~10.7.0' },
npm warn EBADENGINE   current: { node: 'v20.17.0', npm: '10.8.2' }
npm warn EBADENGINE }
```

We should change for: `required: { npm: '>10.7.0' }` in the wet-boew package.json file.

To workaround locally, I needed to run `> npm install npm@10.7.0 -g` to force the use of npm v10.7.0

Next step to fix that issue:
* Update wet-boew and ensure the version indicator is "greater" version
* Add the npm requirement in GCWeb or remove it in wet-boew
* Optional: Only specify the NodeJS requirement in the package.json file

---

This happen because I downgraded my local version of npm.

If during the `script/setup` you get the error `ERESOLVE could not resolve`, just delete all the content in the "node_modules" folder.

**Comment by duboisp** [comment]
@Ricokola FYI, please review the step bellow and let's include those in our release note under "Building GCWeb developer note" sub-title.

Here my step that I took to upgrade my dev build environment on Window 10:

1. Uninstall existing NodeJS version if installed OS wide
2. Get NodeJS LTS recent version
   a. Preferably - Install NVM - Node Version Manager and use Node 20 LTS
   b. Or, install Node 20 LTS OS wide
4. Delete the folder "node-sass" under the "node_modules"
5. Run "script/setup" which would reinstall bower and install all the NodeJS libraries

Then, it will be as usual step to build
1. If wet-boew dependency is updated `> npm install`
2. Build GCWeb `> grunt`
3. Run Jekyll local website `> docker-compose up`
4. Test the webiste + code by navigating locally: http://localhost:4000/ 

---

There is an error in wet-boew, we need to change how we enforce the npm dependencies
```
npm warn EBADENGINE Unsupported engine {
npm warn EBADENGINE   package: 'wet-boew@4.0.81.3',
npm warn EBADENGINE   required: { npm: '~10.7.0' },
npm warn EBADENGINE   current: { node: 'v20.17.0', npm: '10.8.2' }
npm warn EBADENGINE }
```

We should change for: `required: { npm: '>=10.7.0' }` (please check the syntax first) in the wet-boew package.json file.

To workaround locally, I needed to run `> npm install npm@10.7.0 -g` to force the use of npm v10.7.0

Next step to fix that issue:
* Update wet-boew and ensure the version indicator is "greater" version
* Add the npm requirement in GCWeb or remove it in wet-boew
* Optional: Only specify the NodeJS requirement in the package.json file

Note to reinstate from that above error: You will need either to reset the package-lock.json or simply delete it to get it regenerated.

**Comment by Ricokola** [comment]
@Garneauma @duboisp 

Here are the steps for testing locally:

1. Reset hard on this PR latest commit: 716b3f95ee242c4c6ca8c3da74e0f5b8abbf4892
2. Ensure you run node 20 (One solution is installing and using NVM with command: nvm use lts/iron).
3. Execute a "cherry-pick" on the following commit from Eric Dunsworth: 7daac3157469cafdea04a15dc86dbd3a07f044c7
4. Execute a "grunt test" command and then,
5. Build locally using  "docker compose up"

**Review by duboisp — DISMISSED** [review]
Please remove the merge commit

**Review by Garneauma — APPROVED** [review]
I have tested this PR and everything works well locally after the cherry-pick of @EricDunsworth 's PR.

**Review by Garneauma — APPROVED** [review]

**Pull request merged**